### PR TITLE
[xc-admin] add add/remove publishers page

### PR DESCRIPTION
**Pull request by cctdaniel**

### CODE REVIEW

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/UpdatePermissions.tsx [hunk 1]
```diff
@@ -282,6 +284,49 @@ const UpdatePermissions = () => {
...
+                  </div>
+                </>
+              )
+            )
+          })
+        ) : (
+          <p className="mb-8 leading-6">No proposed changes.</p>
+        )}
+        {changes ? (
+          !connected ? (
```
Comment (guibescos) [inline governance/xc-admin/packages/xc-admin-frontend/components/tabs/UpdatePermissions.tsx:312]: This is awesome

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        })
+      })
+      setData(priceAccountToPublishers)
+    }
+  }, [rawConfig, dataIsLoading])
+
+  // function to download json file
+  const handleDownloadJsonButtonClick = () => {
+    const dataStr =
+      'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(data))
```
Comment (guibescos) [inline governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx:82]: `JSON.stringify(data,null,2) `for prettifying

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+  const closeModal = () => {
+    setIsModalOpen(false)
+  }
+
+  useEffect(() => {
+    if (!dataIsLoading && rawConfig) {
+      const priceAccountToPublishers: PriceAccountToPublisherKeys = {}
+      rawConfig.mappingAccounts.map((mappingAccount) => {
+        mappingAccount.products.map((product) => {
+          const priceAccount = product.priceAccounts.find(
```
Comment (guibescos) [inline governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx:65]: JSON key should be the symbol name and not the pubkey

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+        // add instructions to add new publishers
+        publisherKeysToAdd.forEach((publisherKey) => {
+          pythProgramClient.methods
+            .addPublisher(new PublicKey(publisherKey))
+            .accounts({
+              fundingAccount: squads?.getAuthorityPDA(
+                SECURITY_MULTISIG[getMultisigCluster(cluster)],
+                1
+              ),
+              priceAccount: PublicKey.unique(),
```
Comment (guibescos) [inline governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx:177]: typo, make this the actual key

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+        // add instructions to remove publishers
+        publisherKeysToRemove.forEach((publisherKey) => {
+          pythProgramClient.methods
+            .delPublisher(new PublicKey(publisherKey))
+            .accounts({
+              fundingAccount: squads?.getAuthorityPDA(
+                SECURITY_MULTISIG[getMultisigCluster(cluster)],
+                1
+              ),
+              priceAccount: PublicKey.unique(),
```
Comment (guibescos) [inline governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx:191]: Same comment

**Review by guibescos — CHANGES_REQUESTED** [review]
Want to test before approval, please address comments.

File: governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx (hunk 4 above)
Comment (cctdaniel) [reply governance/xc-admin/packages/xc-admin-frontend/components/tabs/AddRemovePublishers.tsx:177]: good catch ser

**Review by guibescos — APPROVED** [review]

**Comment by guibescos** [comment]
Fix reordering bug please

**Pull request merged**